tests/python/relay/test_pass_inline_composites.py

```diff
@@ -0,0 +1,217 @@
+Also, clears Composite and PartionedFromPatterns that infer with certain BYOC implementations
+
+For example suppose we have the graph:
+
+        a  b                   
+        \ /              
+        add     
+         |            
+       relu                            
+
+Merge composite will wrap each standalone op to it's own function, while setting Composite and
+PartitionedFromPattern attrs. 
+       
+Relay IR after merge composite pass when registering each op as a standalone pattern: 
+fn (%a: Tensor[(10, 10), float32], %b: Tensor[(10, 10), float32]) -> Tensor[(10, 10), float32] {
+  %0 = fn (%FunctionVar_0_01: Tensor[(10, 10), float32], %FunctionVar_0_1: Tensor[(10, 10), float32], PartitionedFromPattern="add_", Composite="add") -> Tensor[(10, 10), float32] {
+    add(%FunctionVar_0_01, %FunctionVar_0_1) /* ty=Tensor[(10, 10), float32] */
+  };
+  %1 = %0(%a, %b) /* ty=Tensor[(10, 10), float32] */;
+  %2 = fn (%FunctionVar_0_0: Tensor[(10, 10), float32], PartitionedFromPattern="nn.relu_", Composite="nn.relu") -> Tensor[(10, 10), float32] {
+    nn.relu(%FunctionVar_0_0) /* ty=Tensor[(10, 10), float32] */
+  };
+  %2(%1) /* ty=Tensor[(10, 10), float32] */
+}
+
+Relay IR after inline composites pass:
+fn (%a: Tensor[(10, 10), float32], %b: Tensor[(10, 10), float32]) -> Tensor[(10, 10), float32] {
+  %0 = add(%a, %b) /* ty=Tensor[(10, 10), float32] */;
+  nn.relu(%0) /* ty=Tensor[(10, 10), float32] */
+}
+
+One conventient use of this pass is to use Pattern-based operator support to move away
+from the original operator predicates, and inline them into a single primitive function to offload it 
+to an external BYOC backend, such as TensorRT.
+"""
+
+
+def make_conv_bias_relu_pattern():
+    r"""Create a pattern to match the following graph.
+
+     conv2d
+       |
+    bias_add
+       |
+     relu
+    """
+    x = wildcard()
+    y = wildcard()
+    z = wildcard()
+    conv_node = is_op("nn.conv2d")(x, y)
+    bias_node = is_op("nn.bias_add")(conv_node, z)
+    r = is_op("nn.relu")(bias_node)
+    return r
+
+
+def make_add_relu_pattern():
+    r"""Create a pattern to match the following graph.
+
+     add
+      |
+    relu
+    """
+    add_node = wildcard() + wildcard()
+    r = is_op("nn.relu")(add_node)
+    return r
+
+
+def make_relu_pattern():
+    r"""Create a pattern to match the following graph
+     a
+     |
+    relu
+     |
+    """
+    pattern = is_op("nn.relu")(wildcard())
+    return pattern
+
+
+def make_add_pattern():
+    r"""Create a pattern to match the following graph
+    a  b
+    \  /
+    add
+     |
+    """
+    pattern = is_op("add")(wildcard(), wildcard())
+    return pattern
+
+
+def check_result(pattern_table, graph, expected_graph, import_prelude=False):
+    """Utility function to check inline composites results."""
+    result = run_opt_pass(
+        graph, relay.transform.MergeComposite(pattern_table), import_prelude=import_prelude
+    )
```

Comment:
I think we should put some form of check here just to confirm that a composite function has been created (so we know MergeComposite didn't just skip everything if for instance there was a pattern error).